src/main/scala/com/codacy/parsers/implementation/LCOVParser.scala

```diff
@@ -17,9 +17,10 @@ object LCOVParser extends CoverageParser {
 
   override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] = {
     val report = Try(Source.fromFile(reportFile)) match {
+      case Success(lines) if Try(XMLoader.loadFile(reportFile)).isSuccess =>
```

Comment:
Why? Lcov is not an XML file, it's a weird check.

Reply:
Maybe we should do some schema check instead. Like, all lines conform to the LCOV expected format. Otherwise, this will catch XML files as problematic, but a JSON file will still be analyzed by it

Reply:
Correct, most of the rest of the formats are xml, so, if it's an xml we avoid to try to parse as lcov (this returns `Left`).
In particular LCOVParser was succeeding parsing XML files and was returning `Right` with 0 elements.

In general we should possibly improve this parser to fail "more" if the format is not recognized.

Reply:
I agree with your proposal @franciscodua , but I'm worried I would need to re-write this parser(and possibly introducing bugs/regressions).

In particular I checked a couple of implementations found on the internet:
https://github.com/eriwen/lcov-to-cobertura-xml/blob/master/lcov_cobertura/lcov_cobertura.py#L205
https://github.com/mitchhentges/lcov-rs/blob/master/src/main.rs#L126

and they are very permissive, both examples are swallowing mis-formatted lines (as we do).
I think here we need to find a trade-off ...

Reply:
👍  Agreed.
This is already an improvement. Maybe a bit hard to understand for someone new to the code (a comment might help).
We can think of a better validation in a different issue

Reply:
Do you think the returned error is not explicit enough?
```scala
Left(s"The file is not in the lcov format but is an xml.")
```

Reply:
It is clear that we don't want it to be an xml. But it's not clear why. I won't push on this (it's not that important). I was just thinking along the lines of (most reports are XML, and we want to ensure the LCOV parser won't mishandle it and return an empty result)

Reply:
done